packages/cli/src/s3-state-store.ts

```diff
@@ -4,12 +4,16 @@ import { StateStore } from '@ceramicnetwork/core'
 import LevelUp from 'levelup'
 import S3LevelDOWN from 's3leveldown'
 import toArray from 'stream-to-array'
+import { Semaphore } from 'await-semaphore'
+
+const MAX_CONCURRENT_READS = 4000
```

Comment:
The 5500 limit from S3 is for requests *per second* I assume?  Allowing 4k simultaneous requests could still easily go over the per second limit.  We could do something like we do in pubsub-ratelimit to actually enforce a request per second limit.  I'd also be fine just turning this number down a bit to something like 1000.  

I don't feel like I have a strong instinct on how likely this is to actually bite us in practice.  I'd be interested in @oed and @smrz2001's opinions here

Reply:
I wonder if there's a way to batch requests into one S3 call?

Reply:
@oed, `s3leveldown` does have a batch API but seems like it just makes individual API calls behind-the-scenes. Making direct S3 calls when using this plugin might be complicated.

Reply:
Yeah, @stbrody, I agree, this does seem likely to bust the 5.5K limit. I think it really depends on how S3 measures per-second rates.

Would probably be good to run a quick test with standalone TS (i.e. not from inside Ceramic) calling `s3leveldown` with the semaphore to see when we get throttled. Perhaps a worst case test where we instantly fill up the semaphore and keep it full for a little while.

Alternatively, how hard would it be to use prefixes?

Reply:
The hard part about using prefixes would be the migration.